fix(cmake): make OpenSSL dependency conditional on ssl feature in config

## What

`cmake/network_system-config.cmake.in` unconditionally requires OpenSSL:

```cmake
# Line 11 — always executed, regardless of feature selection
find_dependency(OpenSSL REQUIRED)
```

However, OpenSSL is only an optional `ssl` feature in `vcpkg.json`. A consumer that installs `network_system` **without** the `ssl` feature and then calls `find_package(network_system CONFIG REQUIRED)` will fail with:

```
CMake Error: Could not find a package configuration file provided by "OpenSSL"
```

even though they never requested SSL support.

- **Current**: `find_dependency(OpenSSL REQUIRED)` is always evaluated
- **Expected**: OpenSSL is only required when the `ssl` feature was active at build time

## Why

The unconditional `REQUIRED` forces all consumers to have OpenSSL installed, even for non-TLS network usage (plain TCP, local IPC). This increases the dependency footprint unnecessarily for embedded or constrained environments and contradicts the optional nature of the `ssl` feature.

## Where

- `cmake/network_system-config.cmake.in` — line 11
- `CMakeLists.txt` — the `BUILD_WITH_SSL` CMake option controls whether SSL is compiled in

## How

### Technical Approach

Introduce a build-time flag into the config template, mirroring the pattern used in other kcenon packages (e.g. `monitoring_system-config.cmake.in`):

```cmake
# network_system-config.cmake.in — replace unconditional line 11

set(_NETWORK_WITH_SSL @BUILD_WITH_SSL@)
if(_NETWORK_WITH_SSL)
    find_dependency(OpenSSL REQUIRED)
else()
    find_dependency(OpenSSL QUIET)
endif()
```

`CMakeLists.txt` must pass `BUILD_WITH_SSL` through `configure_package_config_file()` or `configure_file()` so it is baked into the installed config.

### Acceptance Criteria

- [ ] `find_package(network_system)` succeeds on systems without OpenSSL when `ssl` feature was not enabled at build time
- [ ] `find_package(network_system)` still requires OpenSSL when built with `ssl` feature enabled
- [ ] `BUILD_WITH_SSL` variable is correctly substituted into the installed `network_system-config.cmake`
- [ ] No regression for consumers that do use the `ssl` feature

--- a/cmake/network_system-config.cmake.in
+++ b/cmake/network_system-config.cmake.in
@@ -8,7 +8,18 @@ include(CMakeFindDependencyMacro)
 
 # Find required dependencies
 find_dependency(Threads REQUIRED)
-find_dependency(OpenSSL REQUIRED)
+
+# OpenSSL: required when TLS support or WebSocket support (SHA-1 handshake) was enabled
+set(_NETWORK_WITH_TLS @BUILD_TLS_SUPPORT@)
+set(_NETWORK_WITH_WEBSOCKET @BUILD_WEBSOCKET_SUPPORT@)
+if(_NETWORK_WITH_TLS OR _NETWORK_WITH_WEBSOCKET)
+    find_dependency(OpenSSL REQUIRED)
+else()
+    find_dependency(OpenSSL QUIET)
+endif()
+unset(_NETWORK_WITH_TLS)
+unset(_NETWORK_WITH_WEBSOCKET)
+
 find_dependency(ZLIB REQUIRED)
 find_dependency(asio CONFIG REQUIRED)
 

